### ISSUE TITLE
kbd shortcode has accessibility issues

### ISSUE THREAD

**Issue by cwickham** [quarto-dev/quarto-cli#5403]
For JS formats, e.g. HTML, the kbd shortcode sniffs the users OS and inserts icons for the shortcut keys, e.g. `{{< kbd mac=Shift-Command-O >}}` becomes ⇧ ⌘ 0.

Replacing the words with icons is making things much worse for those using screen readers, e.g. (thanks @ateucher and @gtritchie ) for ⇧, ⌘, ⌃, ⌥  NVDA on Windows reads: "upwards hollow arrow", "Mac command key", and says nothing about the Ctrl or the option symbols.

**Comment by cscheid** [comment]
We should definitely fix this, and I can take point, but I could use some help. @ateucher, @gtrichie, do you mind giving me some suggestions here? Thanks!

**Comment by gtritchie** [comment]
The safest approach for screen readers is to use text, so:

Shift
Command 
Option or Opt
Ctrl or Control
PageUp, PageDown
Home, End
Delete
Backspace
Enter, Return

**Comment by cscheid** [comment]
@gtritchie Ok, let me rephrase. Is there really a way to keep the graphical output and indicate to screen readers what the screen-readable content is?

**Comment by gtritchie** [comment]
Let me do some tinkering and I'll reply back here in a bit.

The keyboard shortcut symbols are, I'd argue, not particularly accessible in general, it's not just a concern for screen reader users.

That said, let me see if I can come up with a usable pattern.

**Comment by gtritchie** [comment]
The simplest approach I can come up with is to mark the symbols themselves as hidden from screen readers, and include the text representation adjacent to them, visually hidden, but still in the accessibility tree.

You'll need to define a visually-hidden css class (or can use the one from bootstrap, etc.). Here's a typical example from https://piccalil.li/quick-tip/visually-hidden/

```css
.visually-hidden {
  border: 0;
  clip: rect(0 0 0 0);
  height: auto;
  margin: 0;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
  white-space: nowrap;
}
```

Then, to use the Shift symbol, do this:

```html
<span aria-hidden="true">⇧</span><span class="visually-hidden">Shift</span>
```

This looks exactly the same but will read reliably with any screen reader.

You could also consider adding a tooltip so a mouse-user who isn't familiar with the symbols can hover over it and get a hint, but that's probably overkill (and doesn't help a keyboard-only user).

```html
<span title="Shift" aria-hidden="true">⇧</span><span class="visually-hidden">Shift</span>
```

**Comment by gtritchie** [comment]
An alternative to that is to use images instead of symbol characters, and use `alt="Shift"`. That's what Apple did on this page: https://support.apple.com/en-ca/guide/mac-help/cpmh0011/mac

**Comment by cscheid** [comment]
This is super helpful. I'll give this a shot in the next couple of days and get back to you all. Thanks!

**Comment by mcanouil** [comment]
Maybe of interest: https://github.com/orgs/community/discussions/17926
